src/software/ai/ai.cpp

```diff
@@ -20,39 +20,55 @@ void AI::overridePlay(std::unique_ptr<Play> play)
     override_play = std::move(play);
 }
 
-void AI::overridePlayFromName(std::string name)
+void AI::overridePlayFromProto(TbotsProto::Play play_proto)
 {
-    overridePlay(GenericFactory<std::string, Play, AiConfig>::create(name, ai_config));
+    overridePlay(std::move(createPlay(play_proto, ai_config_)));
+}
+
+void AI::updateAiConfig(TbotsProto::AiConfig ai_config)
+{
+    ai_config_        = ai_config;
+    ai_config_changed = true;
 }
 
 void AI::checkAiConfig()
 {
-    bool current_override = ai_config->getAiControlConfig()->getOverrideAiPlay()->value();
-    std::string current_override_name =
-        ai_config->getAiControlConfig()->getCurrentAiPlay()->value();
-    if (current_override && (current_override != prev_override ||
-                             current_override_name != prev_override_name))
+    auto current_override = ai_config_.ai_control_config().override_ai_play();
+
+    // If we have a new override, and its not back to the Ai selection,
+    // lets override the play
+    if (current_override != prev_override &&
+        current_override != TbotsProto::PlayName::UseAiSelection)
+    {
+        TbotsProto::Play play_proto;
+        play_proto.set_name(current_override);
+        overridePlayFromProto(play_proto);
+    }
+
+    // If we have a new override but its back to the Ai selection, lets
+    // clear the override
+    if (current_override != prev_override &&
+        current_override == TbotsProto::PlayName::UseAiSelection)
     {
-        overridePlayFromName(
-            ai_config->getAiControlConfig()->getCurrentAiPlay()->value());
+        overridePlay(nullptr);
```

Comment:
why do we need both `overridePlayFromProto` and `overridePlay`?

Reply:
There are some instances in simulated play test fixtures that use overridePlay to create a play, update control params and then override. We also call it from the overridePlayFromProto function (which relaces the previous overridePlayFromName)

We use overridePlayFromProto for the simulated playtests